wfl/generate_configs/minim.py

```diff
@@ -201,16 +189,41 @@ def process_step():
                 print('final symmetry group number {}, international (Hermann-Mauguin) {} Hall {} prec {}'.format(
                     dataset['number'], dataset['international'], dataset['hall'], 0.01))
 
-        # Note that if resampling doesn't include original last config, later
-        # steps won't be able to identify those configs as the (perhaps unconverged) minima.
-        # Perhaps status should be set after resampling?
-        traj = _resample_traj(traj, traj_equispaced_n)
 
         if update_config_type:
             # save config_type
             for at0 in traj:
                 config_type_append(at0, at0.info['minim_config_type'])
 
-        all_trajs.append(traj)
+        if traj_subselect is not None:
+            # Note that if resampling doesn't include original last config, later
+            # steps won't be able to identify those configs as the (perhaps unconverged) minima.
+            # Perhaps status should be set after resampling?
+            traj = _resample_traj(traj, subselect=traj_subselect)
+
+        if len(traj) > 0:
```

Comment:
Have you thought about what'll happen when these fail?  Under at least some circumstances the `ConfigSet_out` cleverness tries to match inputs to outputs, and I think it'll behave badly where there's no corresponding output.  Maybe that's OK, and you just can't do the outputs matching inputs in that situation, but it's worth checking. Maybe we should append an empty list, rather than appending nothing?

Also, what happens if _every_ minim fails, so there's no output at all.

Reply:
Good points! `pool.do_in_pool()` actually handles `None`s ([L51](https://github.com/libAtoms/workflow/blob/e654f6fe4a7611a655730877b76114443ba8b0e8/wfl/pipeline/pool.py#L51) & [L155](https://github.com/libAtoms/workflow/blob/e654f6fe4a7611a655730877b76114443ba8b0e8/wfl/pipeline/pool.py#L155)), so let me fix `minim.run_op()` to return `None` for each failed minimisation.

Reply:
Fixed in the last commit.